src/gsy_e/models/area/__init__.py

```diff
@@ -221,22 +235,25 @@ def area_reconfigure_event(self, **kwargs):
                             export_capacity_kVA=export_capacity_kVA
                         )
 
-        except Exception as ex:
+        except (GSyAreaException, GSyDeviceException) as ex:
             log.error(ex)
-            return
+            return None
 
         self._set_grid_fees(grid_fee_constant, grid_fee_percentage)
         self.throughput = throughput
-        self._update_descendants_strategy_prices()
+        self.update_descendants_strategy_prices()
+
+        return None
 
-    def _update_descendants_strategy_prices(self):
+    def update_descendants_strategy_prices(self):
+        """Recursively update the strategy prices of all descendants of the area."""
         try:
             if self.strategy is not None:
                 self.strategy.event_activate_price()
             for child in self.children:
-                child._update_descendants_strategy_prices()
+                child.update_descendants_strategy_prices()
         except Exception:
-            log.exception("area._update_descendants_strategy_prices failed.")
+            log.exception("area.update_descendants_strategy_prices failed.")
```

Comment:
If this is the reason for the change from private to public method, it would be much better to change the exception log instead to something like: `Failed to update strategy prices for the descendants of the area.`